index.js

```diff
@@ -0,0 +1,145 @@
+const express = require('express')
+const bodyParser = require('body-parser')
+const models = require('./models')
+const Op = require('Sequelize').Op
+
+const app = express()
+
+app.get('/movie', async (request, response) => {
+  const movie = await models.movie.findAll({
+    include: [{
+      model: models.genre,
+      as: 'genre'
+    }, {
+      model: models.director,
+      as: 'director'
+    }]
+  })
+  response.send(movie)
+})
+
+app.get('/movie/:identifier', async (request, response) => {
+  const { identifier } = request.params
+  const matchingmovie = await models.movie.findAll({
+    where: {
+      [Op.or]: [{ id: identifier }]
+    },
+    include: [{
+      model: models.genre,
+      as: 'genre'
+    }, {
+      model: models.director,
+      as: 'director'
+    }]
+  })
+
+  if (matchingmovie.length) {
+    response.send(matchingmovie)
+  } else {
+    response.sendStatus(404)
+  }
+})
+
+app.get('/director/:identifier', async (request, response) => {
+  const { identifier } = request.params
+  const matchingdirector = await models.director.findAll({
+    where: {
+      [Op.or]: [{ id: identifier }]
+    },
+    include: [{
+      model: models.movie,
+      as: 'movie'
+    }]
+  })
+
+  if (matchingdirector.length) {
+    response.send(matchingdirector)
+  } else {
+    response.sendStatus(404)
+  }
+})
+app.get('/genre/:gname', async (request, response) => {
+  const { gname } = request.params
+  const matchinggenre = await models.genre.findAll({
+    where: {
+      [Op.or]: [{ name: gname }]
+    },
+    include: [{
+      model: models.movie,
+      as: 'movie'
+    }]
+  })
+
+  if (matchinggenre.length) {
+    response.send(matchinggenre)
+  } else {
+    response.sendStatus(404)
+  }
+})
+app.delete('/movie/:identifier', async (request, response) => {
+  const { identifier } = request.params
+  const matchingmovie = await models.movie.findOne({
+    where: {
+      [Op.or]: [{ id: identifier }]
+    },
+    include: [{
+      model: models.genre,
+      as: 'genre'
+    }, {
+      model: models.director,
+      as: 'director'
+    }]
+  })
+
+  const movieGenres = matchingmovie.getGenre()
+  const movieDirectors = matchingmovie.getDirector()
+  matchingmovie.removeGenre(movieGenres)
+  matchingmovie.removeDirector(movieDirectors)
+  await models.movie.destroy({
+    where: { id: identifier }
+  })
+
+  // If everything goes well respond with the movie
+  return response.status(200).send('movie deleted successfully')
+})
+app.use(bodyParser.json())
+app.post('/movie', async (request, response) => {
+  const { title, director, releaseDate, rated, runTime, genre } = request.body
```

Comment:
You named this `rated` but the example in the README (which I said I'd be testing with) has `rating` so I get an error when testing this.

![image](https://user-images.githubusercontent.com/5775716/70650037-52152180-1c1c-11ea-80a2-d4ccbcb8c852.png)